spec/models/ability_spec.rb

```diff
@@ -1,437 +1,240 @@
 require 'rails_helper'
 require 'cancan/matchers'
 
-RSpec.describe Ability, type: :model do
-  subject { ability }
-  let(:ability) { Ability.new(user) }
+# Run this file with
+#   $ rspec spec/models/ability_spec.rb -fd
+# to see the output of specs running inside the shared examples [mdv]
 
-  context 'ability' do
-    context 'when a user is connected' do
-      let!(:user) { create(:user) }
-      let!(:team) { create(:team) }
+RSpec.describe Ability, type: :model do
+  subject(:ability) { Ability.new(user) }
+  let(:user){ nil }
+  let(:other_user) { create(:user) }
+
+  describe "Guest User" do
```

Comment:
Personal preference and only a suggestion, bud I'd make all of these role-specific `describes` a `context`, such as
> context 'when the user is a guest' do